modules/jpg/image_loader_jpegd.cpp

```diff
@@ -183,6 +180,36 @@ static Vector<uint8_t> _jpgd_buffer_save_func(const Ref<Image> &p_img, float p_q
 	return output;
 }
 
+static Error _jpgd_save_func(const String &p_path, const Ref<Image> &p_img, float p_quality) {
+	ERR_FAIL_COND_V(p_img.is_null() || p_img->is_empty(), ERR_INVALID_PARAMETER);
+	Ref<Image> image = p_img;
+	if (image->get_format() != Image::FORMAT_RGB8) {
+		image->convert(Image::FORMAT_RGB8);
```

Comment:
I see that `_jpgd_buffer_save_func` above does this check instead:
```
	if (image->get_format() != Image::FORMAT_RGB8) {
		image->convert(Image::FORMAT_ETC2_RGB8);
	}
```
Sounds like it's wrong and yours is better? If so should be fixed above too.

I don't know if the intention from #62122 was to do the implementation using `ImageLoaderJPGOSFile` as done here, or just getting a buffer from the file and calling the buffer version.

CC @reduz @Faless

Reply:
I added a `printf` before and inside the if and it seems the image is already RGB8 and the conversion doesn't happen.
If i change the if to 
`
if (image->get_format() != Image::FORMAT_ETC2_RGB8) {
		image->convert(Image::FORMAT_ETC2_RGB8);
	}
`
it gives an error
ERROR: Cannot convert to <-> from compressed formats. Use compress() and decompress() instead.

Reply:
Changed the `_jpgd_buffer_save_func` to use `FORMAT_RGB8`.
Changed the include path to `jpgd.h` and `jpge.h`.
Merged duplicate code into one function.

Can still change the `_jpgd_save_func` to use the buffer from the `_jpgd_buffer_save_func` if need be